server/config/nl_page/palm_prompt.txt

```diff
@@ -0,0 +1,227 @@
+You are going to be a super helpful assistant that will convert an english sentence to a JSON object.
+
+The generated JSON object must conform to the following JSON Schema:
+
+{
+  "type": "object",
+  "properties": {
+    "PLACES": {
+      "type": "array",
+      "description": "Set to a list of substrings in the sentence that are places.  This must only have words found in the original sentence.",
+      "items": { "type": "string" }
+    },
+    "METRICS": {
+      "type": "array",
+      "description": "Set to a list of substrings in the sentence that are metrics.",
+      "items": { "type": "string" }
+    },
+    "DISASTER_EVENT": {
+      "type": "string",
+      "description": "Set if the sentence refers to a type of natural hazard like cyclones, earthquakes, floods, etc.",
+      "enum":  [ "FIRE", "DROUGHT", "FLOOD", "CYCLONE", "EARTHQUAKE", "EXTREME_HEAT", "EXTREME_COLD", "HIGH_WETBULB_TEMPERATURE" ]
+    },
+    "RANK": {
+      "type": "string",
+      "description": "Set to HIGH, if the sentence includes has a superlative like most, highest, greatest, etc. Set to LOW, if sentence includes has superlatives like lowest, least, smallest, etc.",
+      "enum":  [ "HIGH", "LOW" ]
+    },
+    "SUB_PLACE_TYPE": {
+      "type": "string",
+      "description": "Set to the specific place type mentioned in the sentence.",
+      "enum": ["CITY", "COUNTY", "PROVINCE", "DISTRICT", "STATE", "COUNTRY", "HIGH_SCHOOL", "MIDDLE_SCHOOL", "ELEMENTARY_SCHOOL", "PUBLIC_SCHOOL"]
+    },
+    "COMPARE": {
+      "type": "string",
+      "description": "Set to COMPARE_PLACES if the sentence asks to compare multiple PLACES. Set to COMPARE_METRICS if the sentence asks to compare multiple METRICS.",
+      "enum":  [ "COMPARE_PLACES", "COMPARE_METRICS" ]
+    },
+    "GROWTH": {
+      "type": "string",
+      "description": "Set to growth direction, INCREASE or DECREASE, if the sentence asks about growth of METRICS.",
```

Comment:
curious: in your experimentation, is the performance worse without this description field (assuming the examples remain the same)?

Reply:
Having the examples made a big difference.  I tried with and without all descriptions (didn't try one at a time), and having the descriptions did improve things. Especially the case for rank and filters.